src/App.js

```diff
@@ -1,16 +1,144 @@
 import React from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog.js';
+import ColorButtons from './components/ColorButtons.js';
+import { useState } from 'react';
 
 const App = () => {
+  //Used for all changes to message data
+  const [chatBody, setChatBody] = useState(chatMessages);
+
+  //Used for displaying like count
+  const likedMessageCount = chatBody.filter((message) => message.liked).length;
+  const plural = likedMessageCount === 0 || likedMessageCount > 1 ? 's' : '';
+
+  //Used for changes to display of sender names
+  //Later uses assume chat is between exactly two respondents. Local/remote is determined arbitrarily.
+  const uniqueNames = [
+    ...new Set(
+      chatMessages.map((message) => {
+        return message.sender;
+      })
+    ),
+  ];
+  const senders = [uniqueNames[0], uniqueNames[1]];
+  const [senderColors, setSenderColors] = useState(['', '']);
+
+  //Used for header toggles (hidden/visible, night/day mode)
+  const [menuVisibility, setMenuVisibility] = useState('visible');
+  const collapseIcon = menuVisibility === 'visible' ? '🔽' : '🔼';
+  const [dayNightMode, setDayNightMode] = useState('day');
+  const dayNightIcon = dayNightMode === 'day' ? '☀️' : '🌙';
+
+  //Update message body after change in like
+  const updateMessageOnLike = (newMessage) => {
+    const updatedMessages = chatBody.map((message) => {
+      return message.id === newMessage.id ? newMessage : message;
+    });
+    setChatBody(updatedMessages);
+  };
+
+  //Update text color in both header and chat entries
+  const updateMessageColor = (sender, color) => {
+    if (sender === senders[0]) {
+      setSenderColors([color, senderColors[1]]);
+    } else {
+      setSenderColors([senderColors[0], color]);
+    }
```

Comment:
This isn't a complex check (to go from the sender to the position in the color list), but when there's impedence like this, it may be worth switching the data involved. For example, we could store the color infomration as sender→color records in an object. Or we could pass the index of the sender in to the color picker.

Another really effective approach can be to pass in _different_ callback functions to each of the color pickers. Each function would know exactly which record it needs to update. And rather than repeating two very similar functions, we migght write a function that can build each of the other functions, based either on the index or again, the name. JS really likes to make functions!